### PR TITLE
Adding card_viewer_test to test swiping functionality and display of buttons

### LINKED ISSUE #90: Tests for card_viewer
Currently there is no card_viewer_test - we agreed that it would be good to have some: https://github.com/kilikfan/treasure/pull/89#pullrequestreview-600122248. The CardViewer is currently used by the discard page, and also the active quests page as of the linked PR being merged.

### PR DESCRIPTION
**Pull request by samuel-bradley**
Hopefully these tests are valuable - let me know if any more would be useful.

### CODE REVIEW

**Comment by kilikfan** [comment]
Seems sensible to me. Not sure what else you can test as the buttons are constructed by other classes.

**Comment by samuel-bradley** [comment]
> Seems sensible to me. Not sure what else you can test as the buttons are constructed by other classes.

Yes, I think just testing that this renders them is enough here. I think there are some tests that could be added for the button functionality of those other classes but those should probably go in the tests for those classes.

Once this is merged I'll do https://github.com/kilikfan/treasure/issues/91 and add another test for that here.

**Review by ghost — APPROVED** [review]
Only other thing we could test would be the swipe gesture I guess (rather than just tapping on the arrows). But at that point we're basically testing the internals of the `Swiper` gadget that we're using so not really worth it IMO.

File: test/screens/play/card_viewer_test.dart [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import 'package:treasure_of_the_high_seas/model/card/basic/an_island.dart';
+import 'package:treasure_of_the_high_seas/model/card/card.dart' as Model;
+import 'package:treasure_of_the_high_seas/screens/play/card_viewer.dart';
+
+import '../../test_utils.dart';
+
+void main() {
+
+  List<Widget> getButtons(context, card) {
+    return [Text("Test Button 1"), Text("Test Button 2")];
```
Comment (ghost) [inline test/screens/play/card_viewer_test.dart]: I guess we could prove that the `card` argument is being passed correctly if we made use of it here. E.g. we could set button text to "Button 1 for card ${card.name}" or whatever, and check that works :shrug:

Comment (samuel-bradley) [reply test/screens/play/card_viewer_test.dart]: Sure, I'll stick that in later on tonight.

**Pull request merged**